security/pfSense-pkg-snort/files/usr/local/pkg/snort/snort.inc

```diff
@@ -245,19 +228,14 @@ function snort_load_suppress_sigs($snortcfg, $track_by=false) {
 	return $suppress;
 }
 
-/* func builds custom white lists */
+/* func finds custom lists by name and type (defaults to 'whitelist') */
 function snort_find_list($find_name, $type = 'whitelist') {
-	global $config;
-
-	$snortglob = $config['installedpackages']['snortglobal'];
-	if (!is_array($snortglob[$type]))
-		return "";
-	if (!is_array($snortglob[$type]['item']))
-		return "";
 
-	foreach ($snortglob[$type]['item'] as $value) {
-		if ($value['name'] == $find_name)
-			return $value;
+	if (config_get_path("installedpackages/snortglobal/{$type}/item") ) {
+		foreach (config_get_path("installedpackages/snortglobal/{$type}/item") as $value) {
```

Comment:
If you add the default here, you don't need the `if()` test above even, since if it's empty it will be a no-op as the default `[]` will be empty so the `foreach` body would not execute.

```php
foreach (config_get_path("installedpackages/snortglobal/{$type}/item", []) as $value) {
```